src/Microsoft.FeatureManagement/Telemetry/AssignmentReason.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+namespace Microsoft.FeatureManagement.Telemetry
+{
+    /// <summary>
+    /// The reason the variant was assigned during the evaluation of a feature.
+    /// </summary>
+    public enum AssignmentReason
```

Comment:
Not sure if it's just me feeling this confusing. No where in this enum the variant is mentioned. For example, `AssignmentReason.None` sounds like it's saying "something is assigned with no reason". Don't ask me to guess what `AssignmentReason.DisabledDefault` means :) The list is great, but can we come up with something that is more intuitive?

Reply:
We can change the enum name to `VariantAssignmentReason`. Is that enough?

AssignmentReason.None is the default value of the enum. When there is no variant assigned, the reason will be `None`. Maybe, we can change it to `NotAssigned`?

Reply:
`DisabledDefault` is in our feature flag schema. Only when you config the `DisabledDefault` section, you can recieve this reason field in the telemetry.